app/models/accession/README.md

```diff
@@ -0,0 +1,37 @@
+
+In an attempt to reduce the number of execution paths for the accessioning of samples specifically we are going to accession both individual samples and all groups of samples (manifests and studies) using the delayed jobs workers. While this has a number of advantages (explained below), there is currently no method to return feedback to the submitting user. To address this, we are going add database-backed records of the state of each item that is being accessioned.
+
+## Accessioning demand
+
+Looking at accessioning logs for the delayed jobs path over the last month shows that 4 workers process sample accessions in parallel at a combined rate of ~1000 samples/second. While the number of accessions submitted at a time can vary, it is not uncommon to receive requests to process ~5000 samples at a time.
+
+## Multiple APIs
+
+Sequencescape currently uses the original v1 accessioning API provided by EBI (https://ena-docs.readthedocs.io/en/latest/submit/general-guide/webin-v1.html). This has been replaced with a v2 API (https://ena-docs.readthedocs.io/en/latest/submit/general-guide/programmatic.html) which has better support for large submissions.
+
+## Robustness
+
+The delayed job queue allows many accessions to be submitted, but will also retry them should the submission fail. This is particularly helpful if there is an outage on the EBI side.
+
+## Model design
+
+To address the lack of feedback outlined above, a new models and database table are to be created:
+
+- `Accession::SampleStatus` tracks the status and feedback from individual sample accessions
+
+While this is currently intended to be used only with sample accessions, there is no reason why the same pattern could not be used for other accession types in the future.
+
+This is intended to work as follows:
+
+1. A user would submit a sample manifest, accession all samples within a study, or accession individual samples.
+2. Each of these actions would create an individual Accession::SampleStatus record for each item to be accessioned.
+3. If the accessioning process succeeds, the accession number would be updated on the object itself.
+4. If the accessioning process fails, the status record would be updated to 'failed' with the given error message.
+5. All successful statuses would be removed after notification, while failed statuses would be retained for user reference.
+6. Statuses can be deleted after a rolling period of time, say 30 days.
```

Comment:
I am not sure we need to delete them via a cron or manually - they should automatically get deleted if the accession succeeds? I cant imagine a case where we wont re-accession a sample if its required and has so far failed.